src/v_patch.c

```diff
@@ -0,0 +1,380 @@
+//
+// Copyright(C) 2024 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+#include "v_patch.h"
+
+#include "doomtype.h"
+#include "i_printf.h"
+#include "m_swap.h"
+#include "r_defs.h"
+#include "w_wad.h"
+#include "z_zone.h"
+#define M_ARRAY_INIT_CAPACITY 256
+#include "m_array.h"
+
+#include "spng.h"
+
+typedef struct
+{
+    byte row_off;
+    byte *pixels;
+} vpost_t;
+
+typedef struct
+{
+    vpost_t *posts;
+} vcolumn_t;
+
+#define PUTBYTE(r, v)  \
+    *r = (uint8_t)(v); \
+    ++r
+
+#define PUTSHORT(r, v)                              \
+    *(r + 0) = (byte)(((uint16_t)(v) >> 0) & 0xff); \
+    *(r + 1) = (byte)(((uint16_t)(v) >> 8) & 0xff); \
+    r += 2
+
+#define PUTLONG(r, v)                                \
+    *(r + 0) = (byte)(((uint32_t)(v) >> 0) & 0xff);  \
+    *(r + 1) = (byte)(((uint32_t)(v) >> 8) & 0xff);  \
+    *(r + 2) = (byte)(((uint32_t)(v) >> 16) & 0xff); \
+    *(r + 3) = (byte)(((uint32_t)(v) >> 24) & 0xff); \
+    r += 4
+
+//
+// Converts a linear graphic to a patch with transparency. Mostly straight
```

Comment:
I think this code was the reason why we added Simon Judd's copyright when you added DBIGFONT support?